src/runtime/auto-prompt-manager.js

```diff
@@ -365,56 +392,77 @@ export class AutoPromptManager {
    *   dismissedPrompts: (?string|undefined),
    *   shouldShowAutoPrompt: (boolean|undefined),
    * }} params
-   * @return {!string|undefined}
+   * @return {!Promise<string|undefined>}
    */
   getAudienceActionPromptType_({
     article,
     autoPromptType,
     dismissedPrompts,
     shouldShowAutoPrompt,
   }) {
-    let potentialActions = article?.audienceActions?.actions || [];
-    potentialActions = potentialActions.filter((action) =>
-      this.checkActionEligibility_(action.type)
-    );
-
-    // No audience actions means use the default prompt.
-    if (potentialActions.length === 0) {
-      return undefined;
+    const audienceActions = article?.audienceActions?.actions || [];
+
+    const fetchedLocalStorage = /** @type {LocalStorage} */ ({});
+    let promiseChain = Promise.resolve();
+    const actionsIncludeSurvey = audienceActions
+      .map((action) => action.type)
+      .includes(TYPE_REWARDED_SURVEY);
+    if (actionsIncludeSurvey) {
```

Comment:
I would prefer making the logic simpler without having this condition unless local storage reads hinder performance.

```
this.getEvent_(
   completedActionToStorageKey_(
      AnalyticsEvent.ACTION_SURVEY_DATA_TRANSFER
   )
)
.then(surveyCompletions => { 
   let potentialActions = audienceActions.filter((action) =>
   ... 
})
```